pypeit/metadata.py

```diff
@@ -870,9 +866,16 @@ def set_configurations(self, configs=None, force=False, fill=None):
                 if not np.any(indx):
                     continue
                 if metakey is None:
-                    # No matching meta data defined, so just set all
-                    # the frames to this (first) configuration
-                    self.table['setup'][indx] = cfg_key
+                    # No matching meta data defined, so just set all the frames to all of the configurations
+                    new_cfg_key = np.full(len(self.table['setup'][indx]), 'None', dtype=object)
+                    for c in range(len(self.table['setup'][indx])):
+                        if cfg_key in self.table['setup'][indx][c]:
```

Comment:
I'm not sure that this if statement is necessary. `indx` is built using `not_setup` above, so any values in `self.table['setup'][indx][c]` would have been set by  a previous iteration of the loop on 852. So it couldn't contain `cfg_key` as `_configs.keys()` should not have duplicates. But if I'm wrong and it is needed, then it should be followed by an `elif` instead of an `if` on line 874, to prevent `cfg_key` from being added twice by the `else` on lines 876-877.

Reply:
Excellent catch @badpandabear! I had copied this from a line below. This should be an `elif` (and I fixed it in the line below). The first `if` statement catches frames that could be two types (e.g. arc and tilt), and avoids setting it twice. It's pretty rare that this `if` would be used... Thanks!